src/main/java/uk/co/caprica/vlcj/lwjgl/demo/VideoEngineCallbackDemo.java

```diff
@@ -79,7 +79,7 @@
     /**
      * Semaphore used to synchronise the GL context - the context can only be active on one thread at a time.
      */
-    private final Semaphore contextSemaphore = new Semaphore(0);
+    private final Semaphore contextSemaphore = new Semaphore(0, true);
```

Comment:
The true leads into a faster size change for me, as the semaphore is given to the thread waiting the most time already